x-pack/osquerybeat/ext/osquery-extension/main_windows.go

```diff
@@ -7,13 +7,122 @@
 package main
 
 import (
-	"github.com/osquery/osquery-go"
-	"github.com/osquery/osquery-go/plugin/table"
-
+	"github.com/elastic/beats/v7/x-pack/osquerybeat/ext/osquery-extension/pkg/amcache/state"
+	"github.com/elastic/beats/v7/x-pack/osquerybeat/ext/osquery-extension/pkg/amcache/tables"
 	"github.com/elastic/beats/v7/x-pack/osquerybeat/ext/osquery-extension/pkg/browserhistory"
 	"github.com/elastic/beats/v7/x-pack/osquerybeat/ext/osquery-extension/pkg/logger"
+	"github.com/elastic/beats/v7/x-pack/osquerybeat/ext/osquery-extension/pkg/views"
+
+	"github.com/osquery/osquery-go"
+	"github.com/osquery/osquery-go/plugin/table"
 )
 
+func RegisterAmcacheTables(server *osquery.ExtensionManagerServer, log *logger.Logger) {
+	amcacheGlobalState := state.GetAmcacheGlobalState()
+	for _, t := range tables.AllAmcacheTables() {
+		server.RegisterPlugin(table.NewPlugin(string(t.Name), t.Columns(), t.GenerateFunc(amcacheGlobalState, log)))
+	}
+}
+
 func RegisterTables(server *osquery.ExtensionManagerServer, log *logger.Logger) {
 	server.RegisterPlugin(table.NewPlugin("browser_history", browserhistory.GetColumns(), browserhistory.GetGenerateFunc(log)))
+	RegisterAmcacheTables(server, log)
+}
+
+func CreateViews(socket *string, log *logger.Logger) {
```

Comment:
I think we could abstract this a bit, since this is particular to each table, potentially having many more eventually. WDYT about something like

```go
func RegisterTables(server *osquery.ExtensionManagerServer, hooks *Hooks, log *logger.Logger) {
// ...
hooks.RegisterPostHook(amcache.PostHooks())
}
```

or similar, then we can have hooks.PostHooks called in main.go, and each table can choose to register any required hooks and the specific view logic will be part of the amcache package?